Starting FIPS mode in 26.6.1 fails with NoSuchAlgorithmException

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

The configureTruststore Quarkus build step lacks a dependency on CryptoProviderInitBuildItem, making its execution order relative to setCryptoProvider non-deterministic. In 26.5.7, the build step DAG coincidentally scheduled crypto provider initialization first. Changes in 26.6.1 (removal of UserProfileBuildItem steps, new createHttpAccessLogDirectory step, added @Consume(ProfileBuildItem.class) on health/resteasy steps) altered the DAG enough that configureTruststore now executes before the BouncyCastle FIPS provider is registered, causing KeyStore.getInstance("BCFKS") to fail with NoSuchAlgorithmException.

### Version

26.6.1

### Regression

- [x] The issue is a regression

### Expected behavior

FIPS compliant BCFKS works

### Actual behavior

KeyStore.getInstance("BCFKS") fails with NoSuchAlgorithmException

### How to Reproduce?

Start in FIPS mode with 

JAVA_OPTS="$JAVA_OPTS -Djavax.net.ssl.trustStoreType='BCFKS'"

### Anything else?

_No response_

@stianst 

using the default setting doesn't work either, unfortunately 

@slaskawi 's https://github.com/keycloak/keycloak/pull/44506 [hardcodes the usage of the system trust store](https://github.com/keycloak/keycloak/blob/5253082605afc83f865591074fb2f5936185fac3/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/KeycloakRecorder.java#L152) and that causes new issues with FIPS.

This is basically a revival of 

https://github.com/keycloak/keycloak/issues/28526

Which back then was dismissed as "not planned"

So the issue is:

When one sets FIPS to optional, that is:

* BCFIPS are bundled during container build
* KC_FEATURES=fips and relevant JDK flags (e.g. `-Dcom.redhat.fips=true`) are enabled based on the deployment parameters

the start-up will hit the following stack (as bcfips is not used and the code tries to use the normal bc provider)  

```
 ERROR: Failed to start server in (production) mode
  java.lang.RuntimeException: Failed to start quarkus
      at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
      at java.base/jdk.internal.misc.Unsafe.allocateInstance(Native Method)
      at java.base/java.lang.invoke.DirectMethodHandle.allocateInstance(DirectMethodHandle.java:501)
      at io.quarkus.runtime.Quarkus.run(Quarkus.java:79)
      at org.keycloak.quarkus.runtime.KeycloakMain.start(KeycloakMain.java:127)
      at org.keycloak.quarkus.runtime.cli.Picocli.start(Picocli.java:912)
      at org.keycloak.quarkus.runtime.cli.command.AbstractAutoBuildCommand.runCommand(AbstractAutoBuildCommand.java:139)
      at org.keycloak.quarkus.runtime.cli.command.AbstractCommand.lambda$call$1(AbstractCommand.java:66)
      at java.base/java.util.Optional.orElseGet(Optional.java:364)
      at org.keycloak.quarkus.runtime.cli.command.AbstractCommand.call(AbstractCommand.java:65)
      at org.keycloak.quarkus.runtime.cli.command.AbstractCommand.call(AbstractCommand.java:34)
      at picocli.CommandLine.executeUserObject(CommandLine.java:2031)
      at picocli.CommandLine.access$1500(CommandLine.java:148)
      at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2469)
      at picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
      at picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
      at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
      at picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
      at picocli.CommandLine.execute(CommandLine.java:2174)
      at org.keycloak.quarkus.runtime.cli.Picocli.parseAndRun(Picocli.java:180)
      at org.keycloak.quarkus.runtime.KeycloakMain.main(KeycloakMain.java:102)
      at org.keycloak.quarkus.runtime.KeycloakMain.main(KeycloakMain.java:79)
      at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:69)
      at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:37)
  Caused by: java.lang.RuntimeException: Unexpected error when configuring the crypto provider: org.keycloak.crypto.def.DefaultCryptoProvider
      at org.keycloak.quarkus.runtime.KeycloakRecorder.setCryptoProvider(KeycloakRecorder.java:156)
      at io.quarkus.runner.recorded.KeycloakProcessor$setCryptoProvider946368736.deploy_0(Unknown Source)
      at io.quarkus.runner.recorded.KeycloakProcessor$setCryptoProvider946368736.deploy(Unknown Source)
      ... 24 more
  Caused by: java.lang.reflect.InvocationTargetException
      at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:74)
      at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
      at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
      at org.keycloak.quarkus.runtime.KeycloakRecorder.setCryptoProvider(KeycloakRecorder.java:149)
      ... 26 more
  Caused by: java.lang.NoSuchMethodError: 'void org.bouncycastle.crypto.CryptoServicesRegistrar.checkConstraints(org.bouncycastle.crypto.CryptoServiceProperties)'
      at org.bouncycastle.jce.provider.BouncyCastleProvider.loadAlgorithms(Unknown Source)
      at org.bouncycastle.jce.provider.BouncyCastleProvider.setup(Unknown Source)
      at org.bouncycastle.jce.provider.BouncyCastleProvider.access$000(Unknown Source)
      at org.bouncycastle.jce.provider.BouncyCastleProvider$1.run(Unknown Source)
      at java.base/java.security.AccessController.doPrivileged(AccessController.java:319)
      at org.bouncycastle.jce.provider.BouncyCastleProvider.<init>(Unknown Source)
      at org.keycloak.crypto.def.DefaultCryptoProvider.<init>(DefaultCryptoProvider.java:56)
      at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
      ... 29 more
```

Meanwhile, if one leaves FIPS enabled in non strict mode during build, that is:

* BCFIPS is bundled to image
* KC_FEATURE=fips and required JDK flags (e.g. `-Dcom.redhat.fips=true`) are enabled by default
* FIPS_MODE varied in accordance to environment requirements

they will hit:

```
  ERROR: Failed to start server in (production) mode
  java.lang.RuntimeException: Failed to start quarkus
      at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
      at java.base/jdk.internal.misc.Unsafe.allocateInstance(Native Method)
      at java.base/java.lang.invoke.DirectMethodHandle.allocateInstance(DirectMethodHandle.java:501)
      at io.quarkus.runtime.Quarkus.run(Quarkus.java:79)
      at org.keycloak.quarkus.runtime.KeycloakMain.start(KeycloakMain.java:127)
      at org.keycloak.quarkus.runtime.cli.Picocli.start(Picocli.java:912)
      at org.keycloak.quarkus.runtime.cli.command.AbstractAutoBuildCommand.runCommand(AbstractAutoBuildCommand.java:139)
      at org.keycloak.quarkus.runtime.cli.command.AbstractCommand.lambda$call$1(AbstractCommand.java:66)
      at java.base/java.util.Optional.orElseGet(Optional.java:364)
      at org.keycloak.quarkus.runtime.cli.command.AbstractCommand.call(AbstractCommand.java:65)
      at org.keycloak.quarkus.runtime.cli.command.AbstractCommand.call(AbstractCommand.java:34)
      at picocli.CommandLine.executeUserObject(CommandLine.java:2031)
      at picocli.CommandLine.access$1500(CommandLine.java:148)
      at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2469)
      at picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
      at picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
      at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
      at picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
      at picocli.CommandLine.execute(CommandLine.java:2174)
      at org.keycloak.quarkus.runtime.cli.Picocli.parseAndRun(Picocli.java:180)
      at org.keycloak.quarkus.runtime.KeycloakMain.main(KeycloakMain.java:102)
      at org.keycloak.quarkus.runtime.KeycloakMain.main(KeycloakMain.java:79)
      at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:69)
      at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:37)
  Caused by: java.lang.RuntimeException: Failed to initialize truststore: /usr/lib/jvm/java-21-openjdk/lib/security/cacerts, type: PKCS12
      at org.keycloak.truststore.TruststoreBuilder.loadStore(TruststoreBuilder.java:192)
      at org.keycloak.truststore.TruststoreBuilder.includeDefaultTruststore(TruststoreBuilder.java:170)
      at org.keycloak.truststore.TruststoreBuilder.createMergedTruststore(TruststoreBuilder.java:88)
      at org.keycloak.truststore.TruststoreBuilder.setSystemTruststore(TruststoreBuilder.java:54)
      at org.keycloak.quarkus.runtime.KeycloakRecorder.configureTruststore(KeycloakRecorder.java:101)
      at io.quarkus.runner.recorded.KeycloakProcessor$configureTruststore2112467525.deploy_0(Unknown Source)
      at io.quarkus.runner.recorded.KeycloakProcessor$configureTruststore2112467525.deploy(Unknown Source)
      ... 24 more
  Caused by: java.lang.NullPointerException: No password supplied for PKCS#12 KeyStore.
```

BTW, both stacks are dumped while using RHEL with the out of the box FIPS integration.

--- a/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java
+++ b/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java
@@ -343,7 +343,7 @@ class KeycloakProcessor {
     @Record(ExecutionTime.STATIC_INIT)
     @BuildStep
     @Consume(ConfigBuildItem.class)
-    @Consume(CryptoProviderInitBuildItem.class)
+    @Consume(CryptoProviderInitBuildItem.class) // ensures the Providers are loaded prior to handle the keystore #49359
     void configureTruststore(KeycloakRecorder recorder) {
         recorder.configureTruststore();
     }

--- a/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java
+++ b/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java
@@ -343,6 +343,7 @@ class KeycloakProcessor {
     @Record(ExecutionTime.STATIC_INIT)
     @BuildStep
     @Consume(ConfigBuildItem.class)
+    @Consume(CryptoProviderInitBuildItem.class)
     void configureTruststore(KeycloakRecorder recorder) {
         recorder.configureTruststore();
     }

